internal/util/fetch/fetch_test.go

```diff
@@ -433,6 +427,54 @@ func TestCommand_Run_tag(t *testing.T) {
 	})
 }
 
+func TestCommand_Run_subdir_at_tag(t *testing.T) {
+	g := &testutil.TestSetupManager{
+		T: t,
+		ReposChanges: map[string][]testutil.Content{
+			testutil.Upstream: {
+				{
+					Pkg: pkgbuilder.NewRootPkg().
+						WithSubPackages(pkgbuilder.NewSubPkg("java").
+							WithResource("deployment").
+							WithSubPackages(pkgbuilder.NewSubPkg("expected").WithFile("expected.txt", "My kptfile and I should be the only objects"))),
+					Branch: "master",
+					Tag:    "java/v2"},
+			},
+		},
+		LocalChanges: []testutil.Content{
+			{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile(pkgbuilder.NewKptfile().
+						WithUpstreamRef(testutil.Upstream, "/java/expected", "v2", "resource-merge"))},
+		},
+	}
+
+	defer g.Clean()
+	if !g.Init() {
+		return
+	}
+
+	actualPkg := pkgtesting.CreatePkgOrFail(t, g.LocalWorkspace.FullPackagePath())
+	err := Command{
+		Pkg: actualPkg,
+	}.Run(fake.CtxWithNilPrinter())
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	files, err := ioutil.ReadDir(actualPkg.UniquePath.String())
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	fileNames := make([]string, 0)
```

Comment:
I think it might be easier to just use the pkgbuilder to set up the expected content of the fetched package, and then use the `testutil.Assert*` command to verify.

Reply:
Ended up using `AssertEqual` and `AssertKptfile` here.